### PR TITLE
Use global NPM with version check

### PR DESCRIPTION
**Pull request by Turbo87**
... and remove our `npm` dependency.

see https://github.com/ember-cli/rfcs/pull/88

### CODE REVIEW

**Comment by twokul** [comment]
this is nice

File: lib/tasks/npm-task.js [hunk 1]
```diff
@@ -13,47 +20,86 @@ class NpmTask extends Task {
     // The command to run: can be 'install' or 'uninstall'
     this.command = '';
 
-    this.npm = this.npm || require('../utilities/npm');
+    this.versionConstraints = '3 || 4';
+  }
+
+  npm(args) {
+    logger.info('npm: %j', args);
```
Comment (twokul) [inline lib/tasks/npm-task.js]: curious why `%j`? isn't `args` an array?

File: lib/tasks/npm-task.js [hunk 2]
```diff
@@ -13,47 +20,86 @@ class NpmTask extends Task {
     // The command to run: can be 'install' or 'uninstall'
     this.command = '';
 
-    this.npm = this.npm || require('../utilities/npm');
+    this.versionConstraints = '3 || 4';
```
Comment (twokul) [inline lib/tasks/npm-task.js]: would it make sense to turn `versionConstraints` into a `const` on the top of the file or there was some reasoning behind having it as an instance property?

File: lib/tasks/npm-task.js [hunk 3]
```diff
@@ -13,47 +20,86 @@ class NpmTask extends Task {
...
+      throw error;
+    });
   }
 
   run(options) {
-    let ui = this.ui;
-    let startMessage = this.formatStartMessage(options.packages);
-    let completeMessage = this.formatCompleteMessage(options.packages);
+    return this.checkVersion().then(() => {
+
```
Comment (twokul) [inline lib/tasks/npm-task.js]: extra line, intentional?

**Comment by kategengler** [comment]
Assuming I was cc'd for `ember-try`, amusingly, it already relies on there being a global `npm` https://github.com/ember-cli/ember-try/blob/master/lib/dependency-manager-adapters/npm.js#L82

**Comment by rwjblue** [comment]
@kategengler - Ya, likely wanted to ensure communicate that we are going to remove just in case it was setup like bower was before.

**Comment by kategengler** [comment]
@rwjblue Yep, just confirming that ember-try should be fine with this change.

File: lib/tasks/npm-task.js [hunk 4]
```diff
@@ -13,47 +20,86 @@ class NpmTask extends Task {
...
-    this.npm = this.npm || require('../utilities/npm');
+    this.versionConstraints = '3 || 4';
+  }
+
+  npm(args) {
+    logger.info('npm: %j', args);
+    return RSVP.resolve(execa('npm', args, { preferLocal: false }));
+  }
+
+  checkVersion() {
```
Comment (stefanpenner) [inline lib/tasks/npm-task.js]: is this function unit tested?

File: lib/tasks/npm-task.js (hunk 2 above)
Comment (Turbo87) [inline lib/tasks/npm-task.js]: It would be possible, but it seemed more flexible this way.

File: lib/tasks/npm-task.js (hunk 1 above)
Comment (Turbo87) [inline lib/tasks/npm-task.js]: args is an array, yes. %j uses JSON formatting, while the default toString of an array can hide information

File: lib/tasks/npm-task.js (hunk 3 above)
Comment (Turbo87) [inline lib/tasks/npm-task.js]: Yeah, wanted to give the content a little more breathing room

File: lib/tasks/npm-task.js (hunk 4 above)
Comment (Turbo87) [inline lib/tasks/npm-task.js]: not yet, but I guess we could change that

Comment (stefanpenner) [inline lib/tasks/npm-task.js]: with that, I am 👍  with this PR.

Comment (Turbo87) [inline lib/tasks/npm-task.js]: @stefanpenner unit tests for `checkVersion()` are added now

**Comment by stefanpenner** [comment]
@homu r+

**Pull request merged**